packages/@react-spectrum/slider/src/Slider.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {clamp, mergeProps} from '@react-aria/utils';
+import {classNames} from '@react-spectrum/utils';
+import {FocusRing} from '@react-aria/focus';
+import React from 'react';
+import {SliderBase, SliderBaseChildArguments, SliderBaseProps} from './SliderBase';
+import {SpectrumSliderProps} from '@react-types/slider';
+import styles from '@adobe/spectrum-css-temp/components/slider/vars.css';
+import {useHover} from '@react-aria/interactions';
+import {useLocale} from '@react-aria/i18n';
+import {VisuallyHidden} from '@react-aria/visually-hidden';
+
+function Slider(props: SpectrumSliderProps) {
+  let {onChange, value, defaultValue, isFilled, fillOffset, trackGradient, ...otherProps} = props;
+
+  let baseProps: Omit<SliderBaseProps, 'children'> = {
+    ...otherProps,
+    count: 1,
+    // Normalize `value: number[]` to `value: number`
+    value: value != null ? [value] : undefined,
+    defaultValue: defaultValue != null ? [defaultValue] : undefined,
+    onChange(v) {
+      onChange?.(v[0]);
+    }
+  };
+
+  let {direction} = useLocale();
+  let {isHovered, hoverProps} = useHover({});
+
+  return (
+    <SliderBase
+      {...baseProps}
+      classes={{
+        'spectrum-Slider--filled': isFilled && fillOffset == null
+      }}
+      style={
+        // @ts-ignore
+        {'--spectrum-slider-track-color': trackGradient && `linear-gradient(to ${direction === 'ltr' ? 'right' : 'left'}, ${trackGradient.join(', ')})`}
+      }>
+      {({inputRefs: [inputRef], thumbProps: [thumbProps], inputProps: [inputProps], ticks, state}: SliderBaseChildArguments) => {
+        fillOffset = fillOffset != null ? clamp(fillOffset, state.getThumbMinValue(0), state.getThumbMaxValue(0)) : fillOffset;
+
+        let cssDirection = direction === 'rtl' ? 'right' : 'left';
+
+        let lowerTrack = (<div
+          className={classNames(styles, 'spectrum-Slider-track')}
+          style={{
+            width: `${state.getThumbPercent(0) * 100}%`,
+            // TODO not sure if it has advantages, but this could also be implemented as CSS calc():
+            // .track::before {
+            //    background-size: calc((1/ (var(--width)/100)) * 100%);
+            //    width: calc(var(--width) * 1%)M
+            // }
```

Comment:
Interesting idea. Would keep more of it in CSS. For now what you have is ok, but perhaps once we get this into Spectrum CSS they'd want to do it this way?